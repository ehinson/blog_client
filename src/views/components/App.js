import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { createGlobalStyle } from 'styled-components';
import { normalize } from 'styled-normalize';
import { bool } from 'prop-types';

import Registration from 'views/containers/Registration';
import Login from 'views/containers/Login';
import Home from 'views/containers/Home';
import NotFound from 'views/containers/NotFound';
import PrivateRoute from './PrivateRoute';
import PostForm from './Form/PostForm';
import UserForm from './Form/UserForm';
import Posts from './Posts/Posts';
import Post from './Posts/Post';
import Users from './Users/Users';
import User from './Users/User';
import Navigation from 'views/components/Navigation/Navigation';
import Notification from './Notifications/Notification';

const propTypes = {};

const GlobalStyle = createGlobalStyle`
  ${normalize}

  body, html, #root {
    padding: 0;
    margin: 0;
    font-family: 'Montserrat', sans-serif;
    display: flex;
    height: 100%;
    min-height: 100vh;
    flex-direction: column;
    position: relative;
  }
`;

const App = () => {
  return (
    <div>
      <GlobalStyle />
      <Router>
        <Notification />
        <Navigation />

        <Switch>
          <PrivateRoute exact path="/" component={Home} />
          <Route path="/register">
            <Registration />
          </Route>
          <Route path="/login">
            <Login />
          </Route>
          <PrivateRoute path="/users/:id/posts/:post_id/edit" component={PostForm} />
          <PrivateRoute path="/users/:id/posts/add" component={PostForm} />
          <PrivateRoute path="/users/:id/posts/:post_id" component={Post} />
          <PrivateRoute path="/users/:id/posts/" component={Posts} />
          <PrivateRoute path="/users/:id/edit" component={UserForm} />
          <PrivateRoute path="/users/:id/" component={User} />
          <PrivateRoute exact path="/users/" component={Users} />
          <Route path="*" component={NotFound} />
        </Switch>
      </Router>
    </div>
  );
};

App.propTypes = propTypes;

export default App;
