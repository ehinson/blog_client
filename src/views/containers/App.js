import { compose, setPropTypes } from 'recompose';
import { connect } from 'react-redux';
import { func, oneOfType, object } from 'prop-types';

import { getHasUserToken } from '../../state/selectors';

import App from '../components/App';

const propTypes = {
  mapStateToProps: func,
  mapDispatchToProps: oneOfType([func, object]),
};

const mapStateToProps = state => ({
  hasToken: getHasUserToken(state),
});

const mapDispatchToProps = () => ({});

export default compose(setPropTypes(propTypes), connect(mapStateToProps, mapDispatchToProps))(App);
