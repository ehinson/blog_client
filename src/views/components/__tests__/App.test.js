import React from 'react';
import { shallow } from 'enzyme';

// Components
import App from '../App';

describe('App Test Suite', () => {
  it('renders', () => {
    const props = {
      hasToken: true,
    };
    const wrapper = shallow(<App {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
  it('Should match snapshot', () => {
    const props = {
      hasToken: false,
    };
    const wrapper = shallow(<App {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
