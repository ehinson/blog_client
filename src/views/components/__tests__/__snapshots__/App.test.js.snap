// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Suite Should match snapshot 1`] = `
<BrowserRouter>
  <Switch>
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "destroyOnUnmount": true,
              "enableReinitialize": false,
              "forceUnregisterOnUnmount": false,
              "form": "prioritization",
              "getFormState": [Function],
              "initialValues": Object {
                "direction": "desc",
                "sort": "created",
              },
              "keepDirtyOnReinitialize": false,
              "persistentSubmitErrors": false,
              "pure": true,
              "shouldAsyncValidate": [Function],
              "shouldError": [Function],
              "shouldValidate": [Function],
              "shouldWarn": [Function],
              "submitAsSideEffect": false,
              "touchOnBlur": true,
              "touchOnChange": false,
              "updateUnregisteredFields": false,
            },
            "displayName": "ReduxForm",
            "render": [Function],
          },
          "compare": null,
          "displayName": "Connect(ReduxForm)",
          "type": [Function],
        }
      }
      hasToken={false}
      path="/prioritize"
    />
    <Route
      path="/"
    >
      <withRouter(Connect(ReduxForm)) />
    </Route>
  </Switch>
</BrowserRouter>
`;
